src/jit/optcse.cpp

```diff
@@ -2515,18 +2514,16 @@ void Compiler::optValnumCSE_Heuristic()
 
 bool Compiler::optValnumCSE_UnmarkCSEs(GenTree* deadTree, GenTree** wbKeepList)
 {
+    Compiler::fgWalkResult result;
     assert(optValnumCSE_phase);
 
-    // If we have a non-empty *wbKeepList, then we first check for the rare case where we
-    // have a nested CSE def that has side-effects and return false if have this case
+    // First check for the rare case where we have a nested CSE def that has
+    // side-effects and return false whenever we have this case
     //
-    if (*wbKeepList != nullptr)
+    result = fgWalkTreePre(&deadTree, optHasCSEdefWithSideeffect, (void*)wbKeepList);
```

Comment:
I would like not to see `result` redefines and prefer smth like
```
Compiler::fgWalkResult defWithSideEffectResult = fgWalkTreePre(&deadTree, optHasCSEdefWithSideeffect, (void*)wbKeepList);
bool hasPersistentSideEffect = (defWithSideEffectResult == WALK_ABORT);
if (hasPersistentSideEffect)
```

Reply:
OK, I updated it with your suggested changes

Reply:
Thank you, could you also delete `result` that became unused?